packages/@ourworldindata/grapher/src/footer/Footer.tsx

```diff
@@ -10,167 +10,338 @@ import {
         )
+        const licenseAndOriginUrlWidth = Bounds.forText(
+            licenseAndOriginUrlText,
+            { fontSize }
+        ).width
+
+        // If the URL is too long, don't show it
         if (
-            Bounds.forText(finalUrlText, { fontSize: this.fontSize }).width >
-            0.7 * this.maxWidth
+            licenseAndOriginUrlWidth + HORIZONTAL_PADDING >
+            maxWidth - actionButtons.width
         )
             return undefined
-        return finalUrlText
+
+        return correctedUrlText
     }
 
-    @computed private get licenseAndOriginUrlSvg(): string {
-        const { finalUrl, finalUrlText, ccSvg } = this
-        if (!finalUrlText) return ccSvg
+    @computed protected get licenseAndOriginUrlText(): string {
+        const { finalUrlText, licenseText } = this
+        return Footer.constructLicenseAndOriginUrlText(
+            finalUrlText,
+            licenseText
+        )
+    }
 
-        const originUrlLink = `<a target='_blank' style='fill: #777;' href='${finalUrl}'>${finalUrlText}</a>`
-        return [originUrlLink, ccSvg].join(" • ")
+    @computed private get lineHeight(): number {
+        return this.manager.isSmall ? 1.1 : 1.2
     }
 
-    @computed private get fontSize(): number {
-        return 0.7 * (this.manager.fontSize ?? BASE_FONT_SIZE)
+    @computed protected get fontSize(): number {
+        return this.manager.isMedium ? 11 : 12
     }
 
-    @computed private get sources(): MarkdownTextWrap {
-        const { maxWidth, fontSize, sourcesText } = this
-        return new MarkdownTextWrap({
-            maxWidth,
-            fontSize,
-            text: sourcesText,
-        })
+    @computed protected get sourcesFontSize(): number {
+        return this.manager.isSmall ? 12 : 13
     }
 
-    @computed private get note(): MarkdownTextWrap {
-        const { maxWidth, fontSize, noteText } = this
-        return new MarkdownTextWrap({
+    @computed private get hasNote(): boolean {
+        return !!this.noteText
+    }
+
+    @computed private get actionButtonsWidthWithIconsOnly(): number {
+        return new ActionButtons({
+            manager: this.manager,
+            maxWidth: this.maxWidth,
+        }).widthWithIconsOnly
+    }
+
+    @computed private get useFullWidthSources(): boolean {
+        const {
+            hasNote,
+            sourcesFontSize,
             maxWidth,
-            fontSize,
-            text: noteText,
-            detailsOrderedByReference: this.manager
-                .shouldIncludeDetailsInStaticExport
-                ? this.manager.detailsOrderedByReference
-                : new Set(),
-        })
+            sourcesText,
+            actionButtonsWidthWithIconsOnly,
+        } = this
+        if (hasNote) return true
+        const sourcesWidth = Bounds.forText(sourcesText, {
+            fontSize: sourcesFontSize,
+        }).width
+        return sourcesWidth > 2 * (maxWidth - actionButtonsWidthWithIconsOnly)
```

Comment:
Note for the reviewers: This is a heuristic. If the sources text is "kind of long", we render it above the action buttons using the full width. If it's short, it might be placed next to the action buttons.